common/caching/inmemorylru.go

```diff
@@ -0,0 +1,43 @@
+package caching
+
+import (
+	"sync"
+
+	"github.com/golang/groupcache/lru"
```

Comment:
Any reason why we aren't using `github.com/hashicorp/golang-lru` which is what we do for the KeyDB? The Hashicorp version is based on groupcache but also does locking for you so you can remove all the mutex stuff.

Reply:
Not particularly, I've updated it now to use the hashicorp version and removed the mutex stuff.

Reply:
Incidentally, I really don't like that the KeyDB has LRU caching as far down as the database-specific statement functions. At the deepest level it should be at the `Database` layer, and probably sharing the same `InMemoryLRUCache` as room versions.